theano/tensor/nnet/abstract_conv.py

```diff
@@ -177,6 +177,195 @@ def conv2d(input,
     return conv_op(input, filters)
 
 
+def conv2d_grad_wrt_inputs(output_grad,
+                           filters,
+                           output_grad_shape=None,
+                           input_shape=None,
+                           filter_shape=None,
+                           border_mode='valid',
+                           subsample=(1, 1),
+                           filter_flip=True):
+    """This function builds the symbolic graph for getting the
+    gradient of the output of a convolution (namely output_grad)
+    w.r.t the input of the convolution, given a set of 2D filters
+    used by the convolution, such that the output_grad is upsampled
+    to the input shape.
+
+    :type output_grad: symbolic 4D tensor.
```

Comment:
IMO it's quite confusing to call the input of the function `output_grad`. I understand why it was called like this, but probably something like `input_tensor` or 'to_be_upsampled_tensor' would be better.